kitsune/products/models/pages.py

```diff
@@ -0,0 +1,154 @@
+from django.db import models
+from django.http import Http404
+
+from kitsune.products.models import Product
+from kitsune.wiki.facets import topics_for
+from kitsune.wiki.utils import get_featured_articles
+
+from typing import List
+
+from wagtail import blocks
+from wagtail.fields import StreamField
+from wagtail.admin.panels import FieldPanel
+from wagtail.models import Page
+from wagtail.snippets.blocks import SnippetChooserBlock
+
+
+class SumoPlaceholderPage(Page):
+    """A page used to allow for child pages to be created
+    so we can have a proper Wagtail tree structure"""
+
+    settings_panels = Page.settings_panels + [
+        FieldPanel("show_in_menus"),
+    ]
+    content_panels = [
+        FieldPanel("title"),
+        FieldPanel("slug"),
+    ]
+
+    promote_panels: List[FieldPanel] = []
+    preview_modes: List[Page.preview_modes] = []
+
+    is_placeholder = True
+
+    def serve(self, request):
+        return Http404
+
+
+# Define Blocks for Stream Fields
+# Wagtail: This is a StructBlock that allows selection of a Product Snippet
+class ProductSnippetBlock(blocks.StructBlock):
+    """Block for product snippets"""
+
+    product = SnippetChooserBlock(target_model="products.Product", required=True)
+
+    class Meta:
+        template = "products/blocks/product_snippet_block.html"
+        icon = "placeholder"
+        label = "Product Card"
+
+
+class SearchBlock(blocks.StructBlock):
+    """Block for the search form"""
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        return context
+
+    class Meta:
+        template = "products/blocks/search_block.html"
+        icon = "search"
+        label = "Search Form"
+
+
+class CTABlock(blocks.StructBlock):
+    """Block for the call to action"""
+
+    # Doesn't do much at the moment...#todo
+
+    headline = blocks.CharBlock(required=True, max_length=255)
+    details = blocks.RichTextBlock(required=True)
+    link = blocks.URLBlock(required=True)
+    type = blocks.ChoiceBlock(
+        choices=[
+            ("Community", "Community"),
+            ("Paid", "Paid"),
+            ("Other", "Other"),
+        ]
+    )
+
+    class Meta:
+        template = "products/blocks/cta_block.html"
+        icon = "plus-inverse"
+        label = "Call to Action"
+
+
+class FeaturedArticlesBlock(blocks.StructBlock):
```

Comment:
We should discuss how we want this to work at a greater level of detail. 
You can either add singular featured articles, or you can add the default 'featured_articles' block. Or both. So having the separate blocks gives some flexibility. 
However, we could choose not to do that, or we could choose some other way to populate the featured articles, etc. 
Also, since featured articles won't populate locally, it was little bit of a "I need to see things show up" hack to add the singular article selection.

Reply:
Would something like the following work? In principle, I am not against separating them if/when needed. It just feels more DRY to bundle the functionality about featured articles in one block unless there's a specific need that will prevent us from doing so

```python
class FeaturedContentBlock(blocks.StructBlock):
    """Block for featured content, either multiple articles or a single article"""

    content_type = blocks.ChoiceBlock(
        choices=[
            ("multiple", "Multiple Articles"),
            ("single", "Single Article"),
        ],
        required=True,
    )
    articles = blocks.ListBlock(
        SnippetChooserBlock(target_model="wiki.Document"),
        required=False,
        help_text="Select multiple articles if 'Multiple Articles' is chosen",
    )
    article = SnippetChooserBlock(
        target_model="wiki.Document",
        required=False,
        help_text="Select a single article if 'Single Article' is chosen",
    )

    def get_context(self, value, parent_context=None):
        context = super().get_context(value, parent_context=parent_context)
        if value["content_type"] == "multiple":
            context["featured"] = get_featured_articles(product=context["product"], locale="en-US")
        else:
            context["featured_article"] = value["article"]
        return context
```

Reply:
This will have a usability issue as it stands; both `articles` and `article` will appear as available fields in the admin. So even though we provide `help_text` people can still just add whatever to whichever field. I think the single article selection is a bit dodgy right now anyway, since the modal will show you a paginated listing of all documents at the moment, with no facet or search.
There is a way to inject JS into the editor with a hook (` @hooks.register("insert_editor_js" `) so we could show/hide editor fields that way, but there isn't another way to hide fields based on other fields in the admin that I'm aware of.
I think I'd prefer only having the ability to add single articles, up to a specific number. But that means we have to either let free-form selection by the content/admin person, or we present only limited choices.
Thoughts? I'll take this question to Slack as well.